src/components/Panel/Panel.tsx

```diff
@@ -37,45 +37,48 @@ const errorBackgroundColor = addOpacityToHsla(red500, 0.05);
 export function Panel() {
   const { content, page, panelOpen, closePanel } = usePanelContext();
   const [inputValue, setInputValue] = useState("");
-  const { actions } = useActions();
 
   const {
     chainId,
     oracleType,
     title,
     project,
-    ancillaryData,
-    decodedAncillaryData,
+    valueText,
+    queryText,
+    queryTextHex,
     timeUNIX,
     timeUTC,
-    price,
-    assertion,
-    currency,
     formattedBond,
     formattedReward,
     formattedLivenessEndsIn,
     actionType,
-    action,
     expiryType,
     moreInformation,
-    error,
-    setError,
   } = content ?? {};
 
+  const [error, setError] = useState("");
+  const {
+    fetchCurrencyBalance,
+    fetchCurrencyAllowance,
+    token,
+    fetchCurrencyTokenInfo,
+  } = useComputed(content);
```

Comment:
this might be able to evnetually be improved by including these functions in the content, so its scoped with the panel data, rather than being globally computed.  nothing you need to change now but just thinking about this